src/Shared/Microsoft.DotNet.Kusto/KustoClientProvider.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Kusto.Data.Common;
+using Kusto.Data.Net.Client;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.DotNet.Kusto
+{
+    public class KustoClientProvider : IKustoClientProvider
+    {
+        private readonly IOptions<KustoClientProviderOptions> _options;
+
+        public KustoClientProvider(IOptions<KustoClientProviderOptions> options)
+        {
+            _options = options;
+        }
+
+        public KustoClientProviderOptions Options => _options.Value;
+
+        public ICslQueryProvider GetKustoQueryConnectionProvider()
+        {
+            return KustoClientFactory.CreateCslQueryProvider(Options.QueryConnectionString);
```

Comment:
We should probably cache this client and always return the same one.  It's fine to use the same client in multiple threads, I think, so we might as well just hold on to the one we have.  See [here](https://docs.microsoft.com/en-us/azure/kusto/api/netfx/about-kusto-data) where they recommend doing exactly that in their docs

We could be really fancy and use "IOptionsMonitor<KustoClientProviderOptions>" and clear the cache if the value changes. :-)